code/game/verbs/suicide.dm

```diff
@@ -1,84 +1,52 @@
+/mob/living/verb/suicide()
 	set hidden = 1
 
 	if (stat == 2)
 		to_chat(src, "You're already dead!")
-		return
-
+		return FALSE
 	if (!ticker)
 		to_chat(src, "You can't commit suicide before the game starts!")
-		return
+		return FALSE
 
 	if (suiciding)
 		to_chat(src, "You're already committing suicide! Be patient!")
-		return
+		return FALSE
+
+	return (alert("Are you sure you want to commit suicide?", "Confirm Suicide", "Yes", "No") == "Yes")
+
 
-	var/confirm = alert("Are you sure you want to commit suicide?", "Confirm Suicide", "Yes", "No")
+/mob/living/carbon/human/suicide() /// At best, useful for admins to see if it's being called.
+	..()
+	to_chat(src, "<span class='warning'>No. Adminhelp if there is a legitimate reason, and please review our server rules.</span>")
+	message_admins("[ckey] has tried to trigger the suicide verb as human, but it is currently disabled.")
 
-	if(confirm == "Yes")
+/mob/living/carbon/brain/suicide()
+	if(..())
 		suiciding = 1
-		to_chat(viewers(loc),"<span class='danger'>[src]'s brain is growing dull and lifeless. It looks like it's lost the will to live.</span>")
+		visible_message("<span class='danger'>[src]'s brain is growing dull and lifeless. It looks like it's lost the will to live.</span>")
 		spawn(50)
 			death(0)
 			suiciding = 0
 
-/mob/living/silicon/ai/verb/suicide()
-	set hidden = 1
-
-	if (stat == 2)
-		to_chat(src, "You're already dead!")
-		return
-
-	if (suiciding)
-		to_chat(src, "You're already committing suicide! Be patient!")
-		return
-
-	var/confirm = alert("Are you sure you want to commit suicide?", "Confirm Suicide", "Yes", "No")
-
-	if(confirm == "Yes")
+/mob/living/silicon/ai/suicide()
+	if(..())
 		suiciding = 1
-		to_chat(viewers(src),"<span class='danger'>[src] is powering down. It looks like they're trying to commit suicide.</span>")
+		visible_message("<span class='danger'>[src] is powering down. It looks like they're trying to commit suicide.</span>")
 		//put em at -175
 		adjustOxyLoss(max(getMaxHealth() * 2 - getToxLoss() - getFireLoss() - getBruteLoss() - getOxyLoss(), 0))
 		updatehealth()
 
-/mob/living/silicon/robot/verb/suicide()
-	set hidden = 1
-
-	if (stat == 2)
-		to_chat(src, "You're already dead!")
-		return
-
-	if (suiciding)
-		to_chat(src, "You're already committing suicide! Be patient!")
-		return
-
-	var/confirm = alert("Are you sure you want to commit suicide?", "Confirm Suicide", "Yes", "No")
-
-	if(confirm == "Yes")
+/mob/living/silicon/robot/suicide()
+	if(..())
 		suiciding = 1
-		to_chat(viewers(src),"<span class='danger'>[src] is powering down. It looks like they're trying to commit suicide.</span>")
+		visible_message("<span class='danger'>[src] is powering down. It looks like they're trying to commit suicide.</span>")
 		//put em at -175
 		adjustOxyLoss(max(getMaxHealth() * 2 - getToxLoss() - getFireLoss() - getBruteLoss() - getOxyLoss(), 0))
 		updatehealth()
 
-/mob/living/silicon/pai/verb/suicide()
+/mob/living/silicon/pai/suicide()

```

Comment:
This one sort of has to be separate due to the pAI reimplementing the verb categories etc. Personally I'd rename it to Wipe Core or something instead of suicide.

Reply:
pAIs also apparently have `/mob/living/silicon/pai/verb/wipe_software()`, which seems to do exactly this.

Reply:
I thought they did! Didn't go and check so didn't want to make a false claim.